validation/provisioning/rke2/cloud_provider_test.go

```diff
@@ -184,3 +184,65 @@ func TestVSphereCloudProvider(t *testing.T) {
 		}
 	}
 }
+
+func TestHarvesterCloudProvider(t *testing.T) {
+	t.Parallel()
+	r := cloudProviderSetup(t)
+
+	nodeRolesDedicated := []provisioninginput.MachinePools{provisioninginput.EtcdMachinePool, provisioninginput.ControlPlaneMachinePool, provisioninginput.WorkerMachinePool}
+	nodeRolesDedicated[0].MachinePoolConfig.Quantity = 1
+	nodeRolesDedicated[1].MachinePoolConfig.Quantity = 2
+	nodeRolesDedicated[2].MachinePoolConfig.Quantity = 2
```

Comment:
Do we want the pools to be configurable?

Reply:
not for this one, for now I'm doing a minimal deployment with etcd compared to the other 2 cloud provider tests. For these tests its ideal to have more than 1 worker and cp to make sure scheduling by default is working properly